pkg/TIPA.py

```
@@ -158,593 +185,527 @@ class MessageFunc():
-    def run(self, lock):
+        # The debug mode determines what logs and images are shown when running
+        logger_levels = {
+            10: 3,  # DEBUG
+            20: 1,  # INFO
+            30: 2,  # WARNING
+            40: 2,  # ERROR
+            50: 2,  # CRITICAL
+        }
+        self.debug_mode = logger_levels[logger.level]
+
+    def run(self, lock: LockType) -> None:
```

Ruff (0.8.0)

`run` is too complex (20 > 10)

(C901)